run_speech_recognition_whisper.py

```diff
@@ -616,29 +732,42 @@ def is_labels_non_zero(transcription):
         model.freeze_encoder()
         logging.info("Model encoder has been frozen")
 
-    # now that we have our model and tokenizer defined, we can tokenize the text data
-    whisper_tok = whisper.tokenizer.get_tokenizer(False, task="transcribe", language="en")
-    decoding_options = whisper.decoding.DecodingOptions(task="transcribe", language="en")
-    task = whisper.decoding.DecodingTask(model, decoding_options)
-    suppress_tokens = task._get_suppress_tokens()
+    # 8. Load Metric
+    #metric_wer = evaluate.load("wer")
+    #metric_cer = evaluate.load("cer")
+    metric_wer = datasets.load_metric("wer")
+    metric_cer = datasets.load_metric("cer")
 
-    logits_processors = [SuppressBlank(whisper_tok), SuppressTokens(suppress_tokens), ApplyTimestampRules(whisper_tok)]
-    tokenizer = whisper_tok.tokenizer
-    tokenizer.pad_token = tokenizer.eos_token
+    def compute_metrics(pred):
+        pred_ids = pred.predictions
+        pred.label_ids[pred.label_ids == -100] = tokenizer.eos_token_id
 
-    def tokenize_transcripts(batch):
-        batch["labels"] = tokenizer(batch[text_column_name]).input_ids
-        return batch
+        pred_str = tokenizer.batch_decode(pred_ids, skip_special_tokens=True)
+        pred_str = [x.lstrip().strip() for x in pred_str]
 
-    vectorized_datasets = vectorized_datasets.map(tokenize_transcripts, num_proc=num_workers,
-                                                  desc="Tokenizing datasets...",
-                                                  remove_columns=next(iter(raw_datasets.values())).column_names)
+        # we do not want to group tokens when computing the metrics
+        label_str = tokenizer.batch_decode(pred.label_ids, skip_special_tokens=True)
 
-    # 8. Load Metric
-    metric_wer = evaluate.load("wer")
-    metric_cer = evaluate.load("cer")
+        wer = metric_wer.compute(predictions=pred_str, references=label_str)
+        cer = metric_cer.compute(predictions=pred_str, references=label_str)
 
-    def compute_metrics(pred):
+        if do_lower_case:
+            pred_str = [x.lower() for x in pred_str]
+            wer_lower = metric_wer.compute(predictions=pred_str, references=label_str)
+            cer_lower = metric_cer.compute(predictions=pred_str, references=label_str)
+        else:
+            wer_lower = None
+            cer_lower = None
+
+        # replace all punctuation in the predicted string
+        normalized_pred_str = [re.sub(r'[,.:;@#?!&$\-\"]+\ *', " ", input_str) for input_str in pred_str]
```

Comment:
Hmm are we really catching all punctuation here?

Reply:
Anything in the `re.sub` will be replaced: https://regex101.com/r/sXlmtD/2

We could use `string.punctuation` and remove all of these terms:
```python
import string
print(string.punctuation)
keep_apostrophe = string.punctuation.replace("'", "")
print(keep_apostrophe)
```